tools/serverpod_cli/lib/src/runner/serverpod_command_runner.dart

```diff
@@ -0,0 +1,101 @@
+import 'package:args/args.dart';
+import 'package:args/command_runner.dart';
+import 'package:pub_semver/pub_semver.dart';
+import 'package:serverpod_cli/src/analytics/analytics.dart';
+import 'package:serverpod_cli/src/downloads/resource_manager.dart';
+import 'package:serverpod_cli/src/generated/version.dart';
+import 'package:serverpod_cli/src/logger/logger.dart';
+import 'package:serverpod_cli/src/shared/environment.dart';
+import 'package:serverpod_cli/src/update_prompt/prompt_to_update.dart';
+import 'package:serverpod_cli/src/util/exit_exception.dart';
+
+abstract class GlobalFlags {
+  static const developmentPrint = 'development-print';
+}
+
+class ServerpodCommandRunner extends CommandRunner {
+  final Analytics _analytics;
+  final bool _productionMode;
+  ServerpodCommandRunner(
+    this._analytics,
+    this._productionMode,
+    super.executableName,
+    super.description,
+  ) {
+    argParser.addFlag(
+      GlobalFlags.developmentPrint,
+      defaultsTo: true,
+      negatable: true,
+      help: 'Prints additional information useful for development.',
+    );
+  }
+
+  static ServerpodCommandRunner createCommandRunner(
+    Analytics analytics,
+    bool productionMode,
+  ) {
+    return ServerpodCommandRunner(
+      analytics,
+      productionMode,
+      'serverpod',
+      'Manage your serverpod app development',
+    );
+  }
+
+  @override
+  ArgResults parse(Iterable<String> args) {
+    try {
+      return super.parse(args);
+    } on UsageException catch (e) {
+      _analytics.track(event: 'invalid');
+      log.error(e.toString(), style: const LogStyle());
+      throw ExitException(ExitCodeType.commandNotFound);
+    }
+  }
+
+  @override
+  Future runCommand(ArgResults topLevelResults) async {
```

Comment:
Typically, it's best practice to specify a type for the Future, e.g. `Future<void>`